automation/src/test/scala/org/broadinstitute/dsde/firecloud/page/workspaces/summary/WorkspaceSummaryPage.scala

```diff
@@ -56,18 +58,61 @@ class WorkspaceSummaryPage(namespace: String, name: String)(implicit webDriver:
     val publishButton = Button("publish-button" inside this)
     val unpublishButton = Button("unpublish-button" inside this)
     val shareWorkspaceButton = Button("share-workspace-button" inside this)
+
+    def clickEdit(): Unit = {
+      editButton.doClick()
+      wsSummaryPage.awaitReady()
+    }
+
+    def clickSave(): Unit = {
+      saveButton.doClick()
+      wsSummaryPage.awaitReady()
+    }
+
+    def clickCancel(): Unit = {
+      cancelButton.doClick()
+      wsSummaryPage.awaitReady()
+    }
+
+    def clickClone(): CloneWorkspaceModal = {
+      cloneButton.doClick()
+      await ready new CloneWorkspaceModal()
+    }
+
+    def clickDeleteWorkspace(): DeleteWorkspaceModal = {
+      deleteWorkspaceButton.doClick()
+      await ready new DeleteWorkspaceModal
+    }
+
+    def clickPublish(): Unit = {
+      publishButton.doClick()
+      wsSummaryPage.awaitReady()
+    }
+
+    def clickUnpublish(): MessageModal = {
+      unpublishButton.doClick()
+      await ready new MessageModal
+    }
+
+    def clickShareWorkspaceButton(): AclEditor = {
+      shareWorkspaceButton.doClick()
+      await ready new AclEditor()
+    }
   }
 
   private val workspaceAttributesArea = Collapse("attribute-editor", new FireCloudView {
     override def awaitReady(): Unit = {
       table.awaitReady()
-      awaitState("enabled")
+      wsSummaryPage.awaitReady()
     }
 
     val newButton = Button("add-new-button")
     val table = Table("workspace-attributes")
 
-    def clickNewButton(): Unit = newButton.doClick()
+    def clickNewButton(): Unit = {
+      newButton.doClick()
+      awaitReady()
```

Comment:
This is `awaitReady` on the `Collapse`? I wouldn't expect that clicking this button affects the `Collapse` component itself, rather the content inside the `Collapse`. Looks like we could wait on the key input with a `data-test-id` of `-key`.

Reply:
no, it's not `awaitReady` from `Collapse`. It's calling `workspaceAttributesArea.awaitReady` defined at line# 104.